pkg/client/typed/dynamic/client.go

```diff
@@ -177,6 +189,14 @@ func (rc *ResourceClient) Watch(opts v1.ListOptions) (watch.Interface, error) {
 		Watch()
 }
 
+// UnversionedWatch returns a watch.Interface that watches the resource.
+func (rc *ResourceClient) UnversionedWatch(opts api.ListOptions) (watch.Interface, error) {
```

Comment:
And anyway, you should not be passing the unversioned options type to the dynamic client without a codec that correctly interprets it?